token_utils/tests/test_unpack.py

```diff
@@ -0,0 +1,74 @@
+"""
+Tests for token unpacking and verifying.
+"""
+
+import unittest
+from time import time
+
+from jwkest import BadSignature, Expired, Invalid
+
+from token_utils import api
+from token_utils.sign import create_jwt
+from token_utils.unpack import unpack_jwt
+
+test_user_id = 121
+invalid_test_user_id = 120
+test_timeout = 60
+test_now = 1661432902
+test_claims = {"foo": "bar", "baz": "quux", "meaning": 42}
+expected_full_token = {
+    "lms_user_id": test_user_id,
+    "iat": 1661432902,
+    "exp": 1661432902 + 60,
+    "iss": "token-test-issuer",  # these lines from test_settings.py
+    "version": "1.2.0",  # these lines from test_settings.py
+}
+
+
+class TestUnpack(unittest.TestCase):
+    def test_unpack_jwt(self):
+        token = create_jwt(test_user_id, test_timeout, {}, test_now)
+        decoded = unpack_jwt(token, test_user_id, test_now)
+
+        self.assertEqual(expected_full_token, decoded)
+
+    def test_unpack_jwt_with_claims(self):
+        token = create_jwt(test_user_id, test_timeout, test_claims, test_now)
+
+        expected_token_with_claims = expected_full_token.copy()
+        expected_token_with_claims.update(test_claims)
+
+        decoded = unpack_jwt(token, test_user_id, test_now)
+
+        self.assertEqual(expected_token_with_claims, decoded)
+
+    def test_malformed_token(self):
+        token = create_jwt(test_user_id, test_timeout, test_claims, test_now)
+        token = token + "a"
+
+        expected_token_with_claims = expected_full_token.copy()
+        expected_token_with_claims.update(test_claims)
+
+        with self.assertRaises(BadSignature):
+            unpack_jwt(token, test_user_id, test_now)
+
+    def test_unpack_api_hooked_up(self):
+        # todo: is using time() here okay? could there be flakiness?
```

Comment:
Question: is using time() here okay? Could there be flakiness in this test assert? Would having a higher test timeout be better?

Reply:
the reason we use a set test time is to prevent cases where for example the date changes between date creation and date validation in test. In this case it's just checking expiration with a 60s window so probably it can always get to the validation in time (except in the extremely unlikely event of test suspension for 60s) but it's really a good habit just to use a test_now for the token so you know 100% of everything about it like you do for the expired case.

Reply:
I see, yes that makes sense. So I've just included patch to mock the return value for the unpack_jwt function (since that's the call where the time now is included and I figured it would be better not to include this in the api function?)